src/interrupt.rs

```diff
@@ -209,48 +209,56 @@ static mut INTERRUPT_LEVELS: [u128; 8] = [0u128; 8];
 #[ram]
 static INTERRUPT_LEVELS_MUTEX: spin::Mutex<bool> = spin::Mutex::new(false);
 
+#[cfg(feature = "rt")]
```

Comment:
We probably want to cfg away the whole interrupt module if rt is not enabled, unless you have a specific reason for keeping some stuff available?

Reply:
@MabezDev  Originally @reitermarkusmade inclusion of `pub mod timer;` conditional because it uses interrupt::free.
We could use directly from xtensa-lx6, but I chose to reexport for convenience.

@reitermarkus Other option can we do the reexport of only interrupt from the lib.rs file if the rt feature is not selected? So
```
#[cfg(not(feature = "rt"))]
pub use xtensa_lx6::interrupt;
#[cfg(feature = "rt")]
pub mod interrupt;
```

Reply:
I'd prefer not re-exporting `xtensa_lx6::interrupt` as it may lead to confusion when enabling/disabling the rt, feature.

@reitermarkus Could you cfg the entire interrupt module, and remove the feature cfg's you added in there?

Reply:
@MabezDev that was what he did before, but then you need to cfg the timer module as it uses interrupt::free. 

Solution is not to reexport interrupt::free at all and use it directly from xtensa-lx6. Anyway we are moving to using CriticalSectionMutex or CriticalSectionSpinLockMutex with PR #31.

Reply:
I have changed it again to exclude the whole `interrupt` module.